ui/panels-plugin/src/plugins/time-series-table/DataTable.tsx

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/*
+ eslint-disable @typescript-eslint/no-explicit-any
+ */
+import { Fragment, ReactNode } from 'react';
+import { Alert, Table } from '@mui/material';
+import { TimeSeriesData } from '@perses-dev/core';
+import { QueryData } from '@perses-dev/plugin-system';
+
+export interface DataTableProps {
+  result: Array<QueryData<TimeSeriesData>>;
+}
+
+/**
+ * Designed to display timeseries data in a prometheus like table format.
+ * The first column will contain the metric name and label combination, and the second column will contain the values.
+ * This is inspired by prometheus DataTable.
+ * https://github.com/prometheus/prometheus/blob/2524a915915d7eb1b1207152d2e0ce5771193404/web/ui/react-app/src/pages/graph/DataTable.tsx
+ * @param result timeseries query result
+ * @constructor
+ */
+const DataTable = ({ result }: DataTableProps) => {
+  if (!result) {
+    return null;
+  }
+  const series = result.flatMap((d) => d.data).flatMap((d) => d?.series || []);
+  const maxFormattableSize = 1000;
+  let rows: ReactNode[] = [];
+  let limited = false;
+  const doFormat = series.length <= maxFormattableSize;
+  rows = series.map((s, seriesIdx) => {
```

Comment:
Maybe use a dedicated function for readability

Reply:
done